test/libwhisper/whisperTopic.cpp

```diff
@@ -301,4 +302,76 @@ BOOST_AUTO_TEST_CASE(asyncforwarding)
 	BOOST_REQUIRE_EQUAL(result, 1);
 }
 
+BOOST_AUTO_TEST_CASE(topicAdvertising)
+{
+	if (test::Options::get().nonetwork)
+		return;
+
+	cnote << "Testing Topic Advertising...";
+	VerbosityHolder setTemporaryLevel(2);
+
+	Host host1("first", NetworkPreferences("127.0.0.1", 30303, false));
```

Comment:
Can you update this for a random port?
NetworkPreferences p("127.0.0.1");
p.natTraversal = false;

Let me know if I can help here. We eventually need to do this w/all the tests.